pkg/network/usm/sharedlibraries/watcher_test.go

```diff
@@ -519,361 +499,27 @@ func checkPIDAssociatedWithPathID(watcher *soWatcher, pathID pathIdentifier, pid
 	return ok
 }
 
-func checkPIDNotAssociatedWithPathID(watcher *soWatcher, pathID pathIdentifier, pid uint32) bool {
+func checkPIDNotAssociatedWithPathID(watcher *Watcher, pathID PathIdentifier, pid uint32) bool {
 	return !checkPIDAssociatedWithPathID(watcher, pathID, pid)
 }
 
-func createTempTestFile(t *testing.T, name string) (string, pathIdentifier) {
+func createTempTestFile(t *testing.T, name string) (string, PathIdentifier) {
 	fullPath := filepath.Join(t.TempDir(), name)
 
 	f, err := os.Create(fullPath)
+	f.WriteString("foobar")
 	require.NoError(t, err)
 	f.Close()
 	t.Cleanup(func() {
 		os.RemoveAll(fullPath)
 	})
 
-	pathID, err := newPathIdentifier(fullPath)
+	pathID, err := NewPathIdentifier(fullPath)
 	require.NoError(t, err)
 
 	return fullPath, pathID
 }
 
-func checkWatcherStateIsClean(t *testing.T, watcher *soWatcher) {
+func checkWatcherStateIsClean(t *testing.T, watcher *Watcher) {
 	require.True(t, len(watcher.registry.byPID) == 0 && len(watcher.registry.byID) == 0, "watcher state is not clean")
 }
-
-func getTracepointFuncName(tracepointType, name string) string {
-	return fmt.Sprintf("tracepoint__syscalls__sys_%s_%s", tracepointType, name)
-}
-
-const (
-	enterTracepoint = "enter"
-	exitTracepoint  = "exit"
-)
-
-func initEBPFProgram(t *testing.T) *ddebpf.PerfHandler {
```

Comment:
<!-- Praises highlight something positive. Try to leave at least one of these comments per review (if it exists :^)  -->
**👏 praise:** ‏
THANK YOU FOR REMOVING THAT AWFUL CODE